tests/integration/cli/tests/publish.rs

```diff
@@ -0,0 +1,167 @@
+use anyhow::bail;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use wasmer_integration_tests_cli::{get_repo_root_path, get_wasmer_path, ASSET_PATH, C_ASSET_PATH};
+
+fn create_exe_test_wasm_path() -> String {
+    format!("{}/{}", C_ASSET_PATH, "qjs.wasm")
+}
+
+#[test]
+fn wasmer_publish() -> anyhow::Result<()> {
+    // Only run this test in the CI
+    if std::env::var("GITHUB_TOKEN").is_err() {
+        return Ok(());
+    }
+
+    let wapm_dev_token = std::env::var("WAPM_DEV_TOKEN").ok();
+    let tempdir = tempfile::tempdir()?;
+    let path = tempdir.path();
+    let username = "ciuser";
+
+    let random1 = format!("{}", rand::random::<u32>());
+    let random2 = format!("{}", rand::random::<u32>());
+    let random3 = format!("{}", rand::random::<u32>());
+
+    std::fs::copy(create_exe_test_wasm_path(), path.join("largewasmfile.wasm")).unwrap();
+    std::fs::write(
+        path.join("wasmer.toml"),
+        include_str!("./fixtures/init6.toml")
+            .replace("WAPMUSERNAME", username) // <-- TODO!
+            .replace("RANDOMVERSION1", &random1)
+            .replace("RANDOMVERSION2", &random2)
+            .replace("RANDOMVERSION3", &random3),
+    )?;
+
+    let mut cmd = std::process::Command::new(get_wasmer_path());
+    cmd.arg("publish");
+    cmd.arg("--dir");
+    cmd.arg(path);
+    cmd.arg("--quiet");
+    cmd.arg("--registry");
+    cmd.arg("wapm.dev");
+
+    if let Some(token) = wapm_dev_token {
+        cmd.arg("--token");
+        cmd.arg(token);
+    }
+
+    let output = cmd
+        .stdin(Stdio::null())
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
```

Comment:
Note that `Stdio::inherit()` will write *directly* to stdout/stderr, so the output from `wasmer publish` (or any other command we run) will get interleaved with the test runner's output. That could make it annoying to debug failing tests in CI.

Reply:
The integration tests should be run with `--test-threads 1` (running the tests in parallel is relatively useless, since they run very quick). So this will make it possible to have nice output when looking at the logs. I would leave this as is, except for the final test that tests `output.stdout` (we can't use inherit here because we have to capture the stdout).

Reply:
Fixed now, the tests should pass properly since they capture stdout.